File: leetcode/cpp/1309_Decrypt_String_from_Alphabet_to_Integer_Mapping.cpp
#include <string>
class Solution {
public:
	std::string freqAlphabets(std::string s) {
		std::string ret;

		for (int i = 0; i < s.size(); ++i) {
            string pos;
			pos.push_back(s[i]);
			
			if (i + 2 < s.size() && s[i + 2] == '#') {
				pos.push_back(s[i + 1]);
				i += 2;
			}
				
			ret.push_back('a' + std::stoi(pos) - 1);
		}
		return ret;
	}
};

File: leetcode/cpp/1309_Decrypt_String_from_Alphabet_to_Integer_Mapping.py
class Solution:
    def freqAlphabets(self, s: str) -> str:
        ret = []
        i = 0

        while i < len(s):
            if i + 2 < len(s) and s[i + 2] == '#':
                # two-digit number like "10#"
                num = int(s[i:i+2])
                ret.append(chr(ord('a') + num - 1))
                i += 3
            else:
                # single digit
                num = int(s[i])
                ret.append(chr(ord('a') + num - 1))
                i += 1

        return ''.join(ret)
